blackjack.py

```diff
@@ -106,14 +107,18 @@ def get_soft_hand_value(self):
         return (21.1 if self.check_natural_win()
                 else sum(card.points for card in self.hand))
 
-    def get_hard_hand_value(self):
+    def get_hand_value(self):
         """
-        Calculates hand value with Ace as 11
+        Calculates hand value with any Ace as 11 as long as it doesn't lead to bust
         :return: int
         """
-        return (21.1 if self.check_natural_win()
-                else self.get_soft_hand_value() +
-                10 * sum(card.value == 'Ace' for card in self.hand))
+        if self.check_natural_win():
+            return 21.1
+        else:
+            hand_total = self.get_soft_hand_value()
```

Comment:
`get_soft_hand_value` also call `check_natural_win`, but is it necessary? And do we need `get_soft_hand_value` at all?

Reply:
So that thought crossed my mind, and unless I misunderstand the question, here's my thoughts on it..

I see what you mean, get_hand_value will count as many Ace cards as 11 as possible while making sure you don't go bust. 
But in a real life scenario, a player may intend to switch an Ace currently acting like an 11 to act like  a 1. 
For example: 
A player has hand Ace, Ace, 4 (Total: 16) 
With just get_hand_value, one Ace will be treated as 11, the other as 1. 
Now a player might feel ballsy and think "You know, I could hit again, treat both Ace cards as ones and get closer to 21"
Yes, if they hit and it keeps them under 21, the code will correctly treat both Ace cards as 1s. 
But my point is that an actually person player Blackjack might want to know how LOW (6)  their hand value could look, if they wanna take a risk.

Reply:
My main worry is the duplicated code, we should be able to factorize the code instead.
For example, you could do this (also did some reformating):
```
    def get_hand_value(self, soft=False):
        """
        Calculates hand value with any Ace as 11 as long as it doesn't lead to
        bust if soft is False. If soft is True, then Aces are worth 1.
        :return: int
        """
        if self.check_natural_win():
            hand_total = 21.1
        else:
            hand_total = sum(card.points for card in self.hand)
            # Only count Aces as 11 if soft is False
            if (not soft and
                    hand_total <= 11 and
                    any(card.value == 'Ace' for card in self.hand)):
                hand_total += 10
        return hand_total
```

Now if you want to get the hand value, you run:
```
player.get_hand_value()
```
And if you want the soft hand value, you run:
```
player.get_hand_value(soft=True)
```

Note that if you run:
```
player.get_hand_value(soft=False)
```
It will be similar to calling that method without any parameters (because `soft` defaults to `False` in `get_hand_value`).

If you do that, you don't need to have two separates methods with duplicated code anymore :slightly_smiling_face: